specs-jetpack-calypso/wp-jetpack-connect-spec.js

```diff
@@ -0,0 +1,121 @@
+/** @format */
+import test from 'selenium-webdriver/testing';
+import config from 'config';
+import assert from 'assert';
+
+import * as driverManager from '../lib/driver-manager';
+import * as driverHelper from '../lib/driver-helper';
+import * as dataHelper from '../lib/data-helper';
+import { By } from 'selenium-webdriver';
+
+import AddNewSitePage from '../lib/pages/add-new-site-page';
+import PickAPlanPage from '../lib/pages/signup/pick-a-plan-page';
+import SettingsPage from '../lib/pages/settings-page';
+import WporgCreatorPage from '../lib/pages/wporg-creator-page';
+import LoginFlow from '../lib/flows/login-flow';
+import SidebarComponent from '../lib/components/sidebar-component';
+
+const mochaTimeOut = config.get( 'mochaTimeoutMS' );
+const startBrowserTimeoutMS = config.get( 'startBrowserTimeoutMS' );
+const screenSize = driverManager.currentScreenSize();
+
+var driver;
+
+test.before( function() {
+	this.timeout( startBrowserTimeoutMS );
+	driver = driverManager.startBrowser();
+} );
+
+test.describe( `Jetpack Connect: (${ screenSize }) @jetpack`, function() {
+	this.timeout( mochaTimeOut );
+
+	test.describe( 'Connect From Calypso:', function() {
+		this.bailSuite( true );
+
+		test.before( function() {
+			return driverManager.clearCookiesAndDeleteLocalStorage( driver );
+		} );
+
+		test.it( 'Can create wporg site', () => {
+			this.wporgCreator = new WporgCreatorPage( driver );
+			this.wporgCreator.waitForWpadmin();
+		} );
+
+		test.it( 'Can get URL', () => {
+			this.wporgCreator.getUrl().then( url => {
+				this.url = url;
+			} );
+		} );
+
+		test.it( 'Can log in', () => {
+			const loginFlow = new LoginFlow( driver, 'jetpackConnectUser' );
+			loginFlow.loginAndSelectMySite();
+
+			this.sidebarComponent = new SidebarComponent( driver );
+
+			// ensure we are in A/B variant that allows adding jp site from single-site account
+			driver.getCurrentUrl().then( urlDisplayed => {
+				this.sidebarComponent.setABTestControlGroupsInLocalStorage( urlDisplayed );
+			} );
+		} );
+
+		test.it( 'Can disconnect existing jetpack sites', done => {
+			const siteSwitcherSelector = By.css( '.current-site__switch-sites' );
+			const siteSelector = By.css( '.site-selector .site a[aria-label*="ninja"]' );
+
+			const removeSites = () => {
```

Comment:
This whole function can move into `sidebar-component` as `removeAllJetpackSites()` and work by `.is-jetpack` instead of `poopy` or `ninja`.

Reply:
As long as there's a limited number of sites. If that site list grows beyond a threshold and sites are hidden, we would need a search phrase.